mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.java

```diff
@@ -61,12 +63,22 @@ public RecordedRequest(String requestLine, Headers headers, List<Integer> chunkS
       int methodEnd = requestLine.indexOf(' ');
       int pathEnd = requestLine.indexOf(' ', methodEnd + 1);
       this.method = requestLine.substring(0, methodEnd);
-      this.path = requestLine.substring(methodEnd + 1, pathEnd);
+      String path = requestLine.substring(methodEnd + 1, pathEnd);
+      if (!path.startsWith("/")) {
+        path = "/";
```

Comment:
Curious: what's this for? Empty path?

Reply:
CONNECT method.  Maybe I should null out the request URI in this case?  Previously it was because of the HttpUrl.parse, but that didn't appear deliberate.

Thinking about it more, maybe MockWebserver shouldn't fail because it has a bad request path because it seems valid to reproduce a incorrectly implemented server.

Reply:
with a "/" prefix?